dts/msm8916/msm8929-samsung-r04.dts

```diff
@@ -36,6 +36,36 @@
 		};
 	};
 
+	fortuna3g {
+		model = "Samsung Galaxy Grand Prime 3G (SM-G530H)";
+		compatible = "samsung,fortuna3g", "qcom,msm8916", "lk2nd,device";
+		lk2nd,match-bootloader = "G530H*";
+
+		// The stock Samsung bootloader seems to pick <0x0C01FF01 4> as the board id as opposed to <0xCE08FF01 7>
+		// This line should fix that
+		qcom,msm-id = <206 0>;
```

Comment:
```suggestion
		qcom,msm-id = <248 0>;
```

(for 8216, see https://github.com/msm8916-mainline/android_kernel_qcom_msm8916/blob/b20608408caff817ec874f325127b07609fbaeb8/drivers/soc/qcom/socinfo.c#L463)